### PR TITLE
multirotor mission feasibility checks:

### PR DESCRIPTION
**Pull request by tumbili**
make sure that the relative altitude of the takeoff waypoint is
at least one meter higher than the acceptance radius of the waypoint.
This makes sure that the takeoff waypoint is not reached before the vehicle
is at least one meter in the air.

### CODE REVIEW

File: src/modules/navigator/mission_feasibility_checker.cpp [hunk 1]
```diff
@@ -98,7 +98,39 @@ bool MissionFeasibilityChecker::checkMissionFeasible(int mavlink_fd, bool isRota
...
+		if (missionitem.nav_cmd == NAV_CMD_TAKEOFF) {
+			// make sure that the altitude of the waypoint is at least one meter larger than the acceptance radius
+			// this makes sure that the takeoff waypoint is not reached before we are at least one meter in the air
+			float takeoff_alt = missionitem.altitude_is_relative
+				      ? missionitem.altitude
+			              : missionitem.altitude - home_alt;
+			// check if we should use default acceptance radius
+			float acceptance_radius;
+			if (missionitem.acceptance_radius < NAV_EPSILON_POSITION) {
+				acceptance_radius = 10.0f;
```
Comment (LorenzMeier) [inline src/modules/navigator/mission_feasibility_checker.cpp]: There is a call to get the default acceptance radius from the navigator. Never use hardcoded numbers.

Comment (LorenzMeier) [inline src/modules/navigator/mission_feasibility_checker.cpp]: ```get_acceptance_radius();``` in class Navigator.

File: src/modules/navigator/mission_feasibility_checker.cpp [hunk 2]
```diff
@@ -98,7 +98,39 @@ bool MissionFeasibilityChecker::checkMissionFeasible(int mavlink_fd, bool isRota
...
+			} else {
+				acceptance_radius = missionitem.acceptance_radius;
+			}
+
+			if (takeoff_alt - 1.0f < acceptance_radius) {
+				mavlink_log_critical(_mavlink_fd, "Takeoff altitude too low!");
```
Comment (LorenzMeier) [inline src/modules/navigator/mission_feasibility_checker.cpp]: Should say "Mission rejected: Takeoff altitude too low!"

**Comment by tumbili** [comment]
@LorenzMeier Fixed

**Comment by LorenzMeier** [comment]
Is this tested in SITL? If it passes there I would like to merge it. This is something where SITL provides 100% coverage and I can't see why it would be worse than outdoor tests.

**Comment by tumbili** [comment]
@LorenzMeier Tested in SITL but needs one more test after checks passe.

**Comment by LorenzMeier** [comment]
So can we test and merge (locally, not via github) now?

**Comment by tumbili** [comment]
@LorenzMeier Tested it in SITL, works as expected.

**Pull request merged**